File: C/2_find_consecutive_sum_ways.c
#include <stdio.h>

int solution(int n) {
    int answer = 0;
    int k = 1;

    while(k * (k - 1) / 2 < n){
        int kx = n - k * (k - 1) / 2;
        if(kx % k == 0){
            answer ++;
        }
    }
    return answer;
}

int main() {
    int n = 15;
    printf("The number of ways to represent %d as consecutive sums: %d\n", n, solution(n));
    return 0;
}

File: python/2_find_consecutive_sum_ways.py
def solution(n):
    answer = 0
    k = 1

    while k * (k - 1) // 2 < n:
        kx = n - k * (k - 1) // 2
        if kx > 0 and kx % k == 0:
            answer += 1
        k += 1

    return answer

print(solution(15))